### ISSUE TITLE
azurerm_storage_account: from where is 'allow_nested_items_to_be_public' comming?

### ISSUE THREAD

**Issue by J0F3** [hashicorp/terraform-provider-azurerm#27513]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

I like to understand why the property 'allow_nested_items_to_be_public' of a storage account is named this way. 

In every other tooling this setting is called `allowBlobPublicAccess`. So why is it that different in azurem which is not even intuitive to find. What makes it even worse is that one would not find it by searching for "blob" in the documentation because not even the description is a hint to the actual setting in azure: " (Optional) Allow or disallow nested items within this Account to opt into being public."

In additional the default is exactly the opposite as it is actually in Azure. (Default in Azure is `false` where in azurerm it is `true`)

This all makes the setting very difficult to find. Especially when there is an Azure Policy which says you the storage account cannot be created because the setting `allowBlobPublicAccess` must be set to `false` but you cannot find it in Terraform azurerm. 

Azure Portal:
![image](https://github.com/user-attachments/assets/2c493d27-9d82-4c66-81b1-85aaf1db397e)

Azure CLI:
```bash
 az storage account create --allow-blob-public-access $false
```
ARM / Bicep:
```bicep
  properties: {
    allowBlobPublicAccess: false
  ]
```
Azure PowserShell:
```pwsh
New-AzStorageAccount -AllowBlobPublicAccess

```

Terraform azurerm:
```hcl
allow_nested_items_to_be_public = false
```

So, I think it would be worth considering if this cannot be changed to the actual name of the setting and to use the same default value as azure itself is using (false instead of true). 

Thanks!

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Potential Terraform Configuration

```hcl
resource "azurerm_storage_account" "strg" {
  name                = "somestr"
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location

  account_kind             = "StorageV2"
  account_tier             = "Standard"
  account_replication_type = "ZRS"

  allow_blob_public_access = false 
}
```

### References

https://learn.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-configure

**Comment by Chambras** [comment]
@J0F3 I will check if it is available in the current API the provider is using. If it is, might be an easy update. If it is not, we will need to update the API....

**Comment by simonfelding** [comment]
Yeah this makes no sense. [According to Microsoft (and like reported in the description, **it is supposed to be set to `false` by default** ](https://learn.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-overview) but the Terraform provider sets it to `true` by default and uses the wrong name for the key.
